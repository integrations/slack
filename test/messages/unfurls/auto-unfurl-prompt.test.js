const AutoUnfurlPrompt = require('../../../lib/messages/unfurls/auto-unfurl-prompt');

describe('AutoUnfurlPrompt message rendering', () => {
  test('works', async () => {
    const unfurl = {
      githubRepoId: 12345678,
      channelSlackId: 'C12345',
    };
    const team = {
      domain: 'bestslackteam',
    };
    const message = new AutoUnfurlPrompt('integrations', 'snappydoo', unfurl, team);
    expect(message.getAttachment()).toMatchSnapshot();
  });
});
