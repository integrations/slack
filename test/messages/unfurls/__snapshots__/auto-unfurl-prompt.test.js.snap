// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoUnfurlPrompt message rendering works 1`] = `
Object {
  "actions": Array [
    Object {
      "name": "this-channel",
      "style": "primary",
      "text": "Enable for this channel",
      "type": "button",
      "value": "C12345",
    },
    Object {
      "name": "all-channels",
      "text": "Enable for all channels",
      "type": "button",
    },
    Object {
      "name": "cancel",
      "text": "Always ask me",
      "type": "button",
    },
  ],
  "callback_id": "unfurl-auto-12345678|integrations/snappydoo",
  "color": "#24292f",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "You can enable automatic previews for *private* links to integrations/snappydoo that *you* post either to this channel or to all channels in the \`bestslackteam\` workspace.",
  "title": "Done! Want to automatically show rich previews for integrations/snappydoo?",
}
`;
